### PR TITLE
Fix link creation in search in document tool if using TYPO3 speaking urls (slug).

### PR DESCRIPTION
**Pull request by albig**
This would fix probably #747

### CODE REVIEW

**Review by beatrycze-volk — DISMISSED** [review]

**Review by beatrycze-volk — DISMISSED** [review]

File: Resources/Public/Javascript/PageView/SearchInDocument.js [hunk 1]
```diff
@@ -234,15 +256,22 @@ function getNavigationButtons(start, numFound) {
...
         var queryParam = queryParams[i].split('=');
 
         if(decodeURIComponent(queryParam[0]) === $("input[id='tx-dlf-search-in-document-page']").attr('name')) {
             page = parseInt(queryParam[1], 10);
+            pageFound = true;
         }
     }
 
+    if (!pageFound) {
+        var url = baseUrl.split('/');
```
Comment (albig) [inline Resources/Public/Javascript/PageView/SearchInDocument.js:272]: Unfortunately, this code fails at this point: 

```
SearchInDocument.js?1640205096:271 Uncaught ReferenceError: baseUrl is not defined
    at getCurrentPage (SearchInDocument.js?1640205096:271)
    at addImageHighlight (SearchInDocument.js?1640205096:286)
    at Object.success (SearchInDocument.js?1640205096:366)
    at c (jquery-3.5.1.min.js?1605185398:2)
    at Object.fireWith [as resolveWith] (jquery-3.5.1.min.js?1605185398:2)
    at l (jquery-3.5.1.min.js?1605185398:2)
    at XMLHttpRequest.<anonymous> (jquery-3.5.1.min.js?1605185398:2)
```

Comment (albig) [reply Resources/Public/Javascript/PageView/SearchInDocument.js:272]: If I add  `var baseUrl = getBaseUrl(" ");` above, it doesn't failt. But the highlighting isn't shown.

**Review by albig — COMMENTED** [review]
Please have a look at my comments. It doesn't work yet for me :-(

File: Resources/Public/Javascript/PageView/SearchInDocument.js (hunk 1 above)
Comment (beatrycze-volk) [reply Resources/Public/Javascript/PageView/SearchInDocument.js:272]: I have added logger for page. Could you tell me for which page highlighting is performed? (I have no test system with configured slugs, so I'm testing with use of fake links and there it looks correct) I was assuming that problem with highlight after link correction comes from incorrect page number taken by highlight method. This page is checked out by getCurrentPage() method.

File: Resources/Public/Javascript/PageView/SearchInDocument.js [hunk 2]
```diff
@@ -233,16 +255,24 @@ function getNavigationButtons(start, numFound) {
...
+        page = parseInt(url.pop(), 10);
+    }
+    console.log('Page for highlight: ' + page);
```
Comment (albig) [inline Resources/Public/Javascript/PageView/SearchInDocument.js]:
```suggestion
```

**Review by albig — COMMENTED** [review]
This works for me with and without Slug configuration.

Please just remove the console.log().

**Review by beatrycze-volk — APPROVED** [review]

**Pull request merged**